ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLTask.java

```diff
@@ -85,7 +86,24 @@ public int execute() {
       } else {
         throw new IllegalArgumentException("Unknown DDL request: " + ddlDesc.getClass());
       }
-    } catch (Throwable e) {
+    } catch(HiveException e){
+      int excpErrorCode = ReplUtils.handleException(work.isReplication(), e, work.getDumpDirectory(),
```

Comment:
is that a generic `HiveException` handling logic?

Reply:
Yes it is a generic logic. Just to make sure we don't show the stack trace only for this particular case, i have added conditional statement based on the error code.

Reply:
when `work.isReplication() && ReplUtils.shouldIgnoreOnError` you don't need to call handleException